Here is the summary of the results

```
96% tests passed, 42 tests failed out of 1040

Label Time Summary:
2D                =  59.61 sec*proc (19 tests)
3D                =  62.97 sec*proc (15 tests)
aster             =  20.46 sec*proc (7 tests)
battery           =   3.82 sec*proc (1 test)
beam              =   9.77 sec*proc (4 tests)
benchmark         = 116.49 sec*proc (18 tests)
block             = 123.45 sec*proc (37 tests)
circuit_utils     =   2.49 sec*proc (1 test)
circuits          =  95.11 sec*proc (25 tests)
cmodes            = 150.72 sec*proc (24 tests)
conforming        =  24.17 sec*proc (1 test)
contact           =  94.15 sec*proc (24 tests)
control           =  46.75 sec*proc (18 tests)
dg                =   3.54 sec*proc (3 tests)
eigen             =   9.11 sec*proc (3 tests)
elasticity        =  50.11 sec*proc (10 tests)
elasticsolve      = 227.14 sec*proc (45 tests)
eliminate         =   4.17 sec*proc (4 tests)
elmerice          = 685.98 sec*proc (65 tests)
elmerice-fast     =  95.01 sec*proc (36 tests)
elmerice-long     = 355.82 sec*proc (8 tests)
em-wave           =  17.18 sec*proc (5 tests)
extrude           =  21.92 sec*proc (9 tests)
fsi               =  54.31 sec*proc (16 tests)
gauge             =  29.32 sec*proc (3 tests)
harmonic          =  89.55 sec*proc (27 tests)
heateq            = 110.43 sec*proc (47 tests)
helmholtz         =  23.98 sec*proc (9 tests)
homogenization    =   7.33 sec*proc (2 tests)
hutiter           =   2.87 sec*proc (4 tests)
hypre             =  22.58 sec*proc (18 tests)
lua               =  36.04 sec*proc (12 tests)
lumping           =  51.77 sec*proc (20 tests)
matc              = 105.03 sec*proc (35 tests)
mgdyn             = 185.22 sec*proc (44 tests)
mmg               = 134.16 sec*proc (14 tests)
mortar            = 280.04 sec*proc (71 tests)
mumps             =  19.82 sec*proc (9 tests)
n-t               = 162.17 sec*proc (35 tests)
namespace         =  20.68 sec*proc (8 tests)
p-fem             =  85.47 sec*proc (15 tests)
parallel          = 203.55 sec*proc (130 tests)
particle          =  54.98 sec*proc (17 tests)
permafrost        =  18.76 sec*proc (2 tests)
plate             =  30.16 sec*proc (11 tests)
quick             = 1032.95 sec*proc (409 tests)
radiation         =   4.11 sec*proc (1 test)
radiator          =  50.71 sec*proc (8 tests)
resistor          =   2.56 sec*proc (1 test)
restart           =  57.31 sec*proc (21 tests)
rotm              =   6.32 sec*proc (1 test)
scanning          =   8.45 sec*proc (2 tests)
serendipity       = 166.93 sec*proc (36 tests)
serial            = 3295.03 sec*proc (846 tests)
shell             = 216.16 sec*proc (39 tests)
slow              = 801.72 sec*proc (34 tests)
stranded          =   8.64 sec*proc (2 tests)
thin_region       =   8.59 sec*proc (3 tests)
threaded          =  55.62 sec*proc (9 tests)
transient         = 205.88 sec*proc (57 tests)
umat              =   7.54 sec*proc (3 tests)
useextrude        =  57.66 sec*proc (24 tests)
vector_element    =  39.24 sec*proc (14 tests)
viscoelastic      =  25.18 sec*proc (2 tests)
vtu               =  60.46 sec*proc (13 tests)
whitney           = 415.60 sec*proc (79 tests)
wvector           =   6.06 sec*proc (1 test)

Total Test time (real) = 4060.10 sec

The following tests FAILED:
          1 - AIFlowSolve (Failed)
          4 - Calving3D (Failed)
          8 - Damage (Failed)
         20 - FrictionHeatMasked (Failed)
         37 - Hydro_Coupled (Failed)
         38 - Hydro_SedOnly (Failed)
         41 - InvMeth_AdjRobin (Failed)
         43 - MISMIP_FS-SSA (Failed)
         47 - MMG2D_Transient (Failed)
         48 - Permafrost_Biot (Failed)
        194 - CurvedBoundaryCylH_np3 (Failed)
        195 - CurvedBoundaryCylH_np8 (Failed)
        257 - ElastPelem2dPmultgBoomer_np3 (Failed)
        258 - ElastPelem2dPmultgMumps_np3 (Failed)
        364 - InternalPartitioning_np8 (Failed)
        366 - InternalPartitioning2_np6 (Failed)
        368 - InternalPartitioning3_np6 (Failed)
        561 - RotatingBlockRemeshMMG3D (Failed)
        657 - ShoeboxFsiStatic (Failed)
        659 - ShoeboxFsiStaticShell (Failed)
        677 - StatElecSolveMMG3D (Failed)
        679 - StatElecSolveVecMMG3D (Failed)
        740 - WinkelBmNavierHypreBiCGStabBoomer_np4 (Failed)
        746 - WinkelBmPoissonHypreBiCGStabBoomer_np2 (Failed)
        747 - WinkelBmPoissonHypreBiCGStabBoomer_np8 (Failed)
        748 - WinkelBmPoissonHypreBiCGStabIlu0_np2 (Failed)
        749 - WinkelBmPoissonHypreBiCGStabIlu0_np8 (Failed)
        750 - WinkelBmPoissonHypreBiCGStabIlu1_np2 (Failed)
        751 - WinkelBmPoissonHypreBiCGStabIlu1_np8 (Failed)
        752 - WinkelBmPoissonHypreBoomer_np2 (Failed)
        753 - WinkelBmPoissonHypreBoomer_np8 (Failed)
        754 - WinkelBmPoissonHypreCgIlu0_np2 (Failed)
        755 - WinkelBmPoissonHypreCgIlu0_np8 (Failed)
        756 - WinkelBmPoissonHypreFlexGMResILU0_np2 (Failed)
        757 - WinkelBmPoissonHypreFlexGMResILU0_np8 (Failed)
        758 - WinkelBmPoissonHypreGMResBoomer_np2 (Failed)
        759 - WinkelBmPoissonHypreGMResBoomer_np8 (Failed)
        760 - WinkelBmPoissonHypreGMResILU0_np2 (Failed)
        761 - WinkelBmPoissonHypreGMResILU0_np8 (Failed)
        927 - mgdyn_hypre_ams_np2 (Failed)
        940 - mgdyn_steady_quad_extruded_restart (Failed)
        941 - mgdyn_steady_quad_extruded_restart_np3 (Failed)
Errors while running CTest
Output from these tests are in: /root/build/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
make: *** [Makefile:91: test] Error 8
Apptainer> 
```

Quite a few tests fail like this:

```
Apptainer> ctest --rerun-failed --output-on-failure
Test project /root/build
      Start   1: AIFlowSolve
 1/42 Test   #1: AIFlowSolve ..............................***Failed    0.47 sec
-- BINARY_DIR = /root/build
CMake Error at /root/elmerfem/elmerice/Tests/test_macros.cmake:76 (FILE):
  FILE failed to open for reading (No such file or directory):

    /root/build/elmerice/Tests/AIFlowSolve/TEST.PASSED
Call Stack (most recent call first):
  /root/elmerfem/elmerice/Tests/AIFlowSolve/runTest.cmake:3 (RUN_ELMERICE_TEST)
```
And like this, just right after running `Elmergrid`

```
      Start   4: Calving3D
 2/42 Test   #4: Calving3D ................................***Failed    5.41 sec

Starting program Elmergrid, compiled on Jan  5 2024
Elmergrid reading in-line arguments
Lower dimensional boundaries will be removed
Materials and boundaries will be renumbered
Nodes that do not appear in any element will be removed
The mesh will be partitioned with Metis to 4 partitions.
Output will be saved to file PlanMesh.

Elmergrid loading data:
-----------------------
Format chosen using the first line: $MeshFormat
Gmsh version is 2.2
Loading mesh in Gmsh format 2.0 from file PlanMesh.msh
Allocating for 282 knots and 562 elements.
Moving bulk elements to boundary elements
Leading bulk elementtype is 303
Trailing bulk elementtype is 202
There are 66 (out of 562) lower dimensional elements.
Node 77 belongs to maximum of 8 elements
Found correctly 66 side elements.
Parent elements were reordered up to index 496.
Moved 496 elements (out of 562) to new positions
Successfully read the mesh from the Gmsh input file.
Using physical numbering of entities

Elmergrid creating and manipulating meshes:
-------------------------------------------
Removing lower dimensional boundaries
Maximum elementtype is 303 and dimension 2
Minimum elementtype is 303 and dimension 2
Removed 0 (out of 66) less than 1D boundary elements
All 282 nodes were used by the mesh elements
Initial boundary interval [101,104]
BC index changed 101 -> 1 in 13 1D elements
BC index changed 102 -> 2 in 10 1D elements
BC index changed 103 -> 3 in 13 1D elements
BC index changed 104 -> 4 in 30 1D elements
Mapping boundary types from [101 104] to [1 4]
Initial body interval [107,107]
body index changed 107 -> 1 in 496 elements
Mapping material types from [107 107] to [1 1]
Renumbering of material types completed!

Elmergrid partitioning meshes:
------------------------------
Setting default Metis options!
Setting user defined Metis options!
Making a Metis partitioning for 496 elements in 3-dimensions.
All elements are of type 303
Minimum number of linear nodes in elements 3
Requiring number of nodes in dual graph 2
Using all 282 possible nodes in the Metis graph
Allocating mesh topology of size 1488
Starting graph partitioning METIS_PartMeshNodal.
 Runtime parameters:
   Objective type: METIS_OBJTYPE_CUT
   Coarsening type: METIS_CTYPE_SHEM
   Initial partitioning type: METIS_IPTYPE_METISRB
   Refinement type: METIS_RTYPE_GREEDY
   Perform a 2-hop matching: No
   Number of balancing constraints: 1
   Number of refinement iterations: 10
   Random number seed: -1
   Number of partitions: 4
   Number of cuts: 1
   User-supplied ufactor: 30
   Minimize connectivity: No
   Create contigous partitions: No
   Target partition weights: 
        0=[2.50e-01]   1=[2.50e-01]   2=[2.50e-01]   3=[2.50e-01]
   Allowed maximum load imbalance: 1.030 


 gk_mcore statistics
           coresize:         4728         nmops:         2048  cmop:      0
        num_callocs:            9   num_hallocs:            0
       size_callocs:         5840  size_hallocs:            0
        cur_callocs:            0   cur_hallocs:            0
        max_callocs:         2952   max_hallocs:            0
 nbrpool statistics
        nbrpoolsize:            0   nbrpoolcpos:            0
    nbrpoolreallocs:            0


Timing Information -------------------------------------------------
 Multilevel:               0.001
     Coarsening:                   0.000
            Matching:                      0.000
            Contract:                      0.000
     Initial Partition:            0.001
     Uncoarsening:                 0.000
          Refinement:                      0.000
          Projection:                      0.000
     Splitting:                    0.000
********************************************************************

 gk_mcore statistics
           coresize:         4728         nmops:         2048  cmop:      0
        num_callocs:           26   num_hallocs:            0
       size_callocs:         9248  size_hallocs:            0
        cur_callocs:            0   cur_hallocs:            0
        max_callocs:         4576   max_hallocs:            0
 nbrpool statistics
        nbrpoolsize:         1384   nbrpoolcpos:          440
    nbrpoolreallocs:            0

Finished graph partitioning METIS_PartMeshNodal.
Set the partition given by Metis for each node
Successfully made a Metis partition using the element mesh.
Optimizing the partitioning at boundaries.
Ownership of 0 parents was changed at BCs
Optimizing for 4 partitions
Creating a table showing all parenting partitions of nodes.
Nodes belong to 3 partitions in maximum
There are 41 shared nodes which is 14.54 % of all nodes.
The initial owner was not any of the elements for 0 nodes
Checking partitioning before optimization
Checking for partitioning
Information on partition bandwidth
Distribution of elements, nodes and shared nodes
     partition  elements   nodes      shared    
     1          128        72         14        
     2          118        70         8         
     3          123        70         11        
     4          127        70         11        
Average number of elements in partition 7.050e+01
Maximum deviation in ownership 2
Average deviation in ownership 8.660e-01
Average relative deviation 1.23 %
Checking for problematic sharings
Optimizing sharing for 4 partitions
There shouldn't be any problematic sharings, knock, knock...
Partitioning was not altered
Partitioning routines finished!

Elmergrid saving data with method 2:
-------------------------------------
Saving Elmer mesh in partitioned format
Number of boundary nodes at the boundary: 66
Number of additional interface nodes: 0
Reusing existing subdirectory: partitioning.4
Saving mesh in parallel ElmerSolver format to directory PlanMesh/partitioning.4.
Nodes belong to 3 partitions in maximum
Saving mesh for 4 partitions
   part  elements   nodes      shared   bc elems
   1     128        72         14       16      
   2     118        70         8        21      
   3     123        70         11       17      
   4     127        70         11       12      
----------------------------------------------------------------------------------------------
   ave   124.0      70.5       11.0     16.5     0.0     
Writing of partitioned mesh finished

Thank you for using Elmergrid!
Send bug reports and feature wishes to elmeradm@csc.fi
-- BINARY_DIR = /root/build
CMake Warning (dev) at /root/elmerfem/elmerice/Tests/test_macros.cmake:60 (IF):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  Quoted variables like "WITH_MPI" will no longer be dereferenced when the
  policy is set to NEW.  Since the policy is not set the OLD behavior will be
  used.
Call Stack (most recent call first):
  /root/elmerfem/elmerice/Tests/Calving3D/runTest.cmake:35 (RUN_ELMERICE_TEST)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /root/elmerfem/elmerice/Tests/test_macros.cmake:62 (IF):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  Quoted variables like "N" will no longer be dereferenced when the policy is
  set to NEW.  Since the policy is not set the OLD behavior will be used.
Call Stack (most recent call first):
  /root/elmerfem/elmerice/Tests/Calving3D/runTest.cmake:35 (RUN_ELMERICE_TEST)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Error at /root/elmerfem/elmerice/Tests/test_macros.cmake:74 (FILE):
  FILE failed to open for reading (No such file or directory):

    /root/build/elmerice/Tests/Calving3D/TEST.PASSED_4
Call Stack (most recent call first):
  /root/elmerfem/elmerice/Tests/Calving3D/runTest.cmake:35 (RUN_ELMERICE_TEST)
```

--- a/docker/elmer.def
+++ b/docker/elmer.def
@@ -7,6 +7,7 @@ From: debian:12
    export PATH=/opt/elmer/bin:$PATH
    export ELMER_HOME=/opt/elmer
    export ELMER_LIB=/opt/elmer/lib
+   export NCPU=$(nproc)
 
 %post
    # Install only what is asked
@@ -14,7 +15,7 @@ From: debian:12
    apt $APT_OPTIONS update
 
    # Install build tools
-   apt $APT_OPTIONS install build-essential gfortran cmake g++ python3 
+   apt $APT_OPTIONS install build-essential gfortran cmake g++ python3
    # Install download tools
    apt $APT_OPTIONS install git wget
    # Install general set of requirements
@@ -25,7 +26,7 @@ From: debian:12
    # Install required scientific libraries
    apt $APT_OPTIONS install libopenblas64-openmp-dev liblapack64-dev libnetcdff-dev netcdf-bin libcgns-dev libhdf5-openmpi-dev libscotchparmetis-dev libptscotch-dev libmumps64-ptscotch-dev petsc64-dev libhypre64-dev trilinos-all-dev libsuitesparse-dev libumfpack5 libparpack2-dev metis libmetis-dev
    # Install graphical libraries
-   apt $APT_OPTIONS install libqwt-qt5-dev libqt5opengl5-dev qtscript5-dev libqt5svg5-dev 
+   apt $APT_OPTIONS install libqwt-qt5-dev libqt5opengl5-dev qtscript5-dev libqt5svg5-dev
    # Install extra tools
    apt $APT_OPTIONS install lua5.3
    # Install general tools
@@ -36,7 +37,7 @@ From: debian:12
    #################################################
 
    export BASE_PREFIX=/opt/elmer
-   
+
    mkdir -p $BASE_PREFIX
 
    cd $HOME
@@ -53,12 +54,12 @@ From: debian:12
    -DLIBMMG3D_SHARED=ON \
    ..
 
-   make -j14
-   make install -j14
+   make -j$NCPU
+   make install -j$NCPU
    # Clean flags
    export CFLAGS=""
    export CXXFLAGS=""
-   
+
    ## Build ElmerFEM
 
    cd $HOME
@@ -83,13 +84,16 @@ From: debian:12
    -DWITH_MATC=TRUE \
    -DWITH_LUA=TRUE \
    -DWITH_MMG=TRUE \
+   -DEXTERNAL_UMFPACK=TRUE \
+   -DUMFPACK_LIBRARIES="-L/usr/lib/x86_64-linux-gnu/libumfpack.so -lumfpack -lamd -lcholmod -lsuitesparseconfig -lccolamd -lcamd -lbtf" \
+   -DUMFPACK_INCLUDE_DIR="/usr/include/suitesparse" \
    -DCMAKE_INSTALL_PREFIX="$BASE_PREFIX" \
    -DCMAKE_BUILD_TYPE=Release \
    ../elmerfem
 
    # Build and Install
-   make -j14
-   make -j14 install
+   make -j$NCPU
+   make -j$NCPU install
 
 %runscript
    /opt/elmer/bin/ElmerGUI

